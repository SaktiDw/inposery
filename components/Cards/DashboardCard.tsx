import React from "react";
import PriceFormater from "../FormatText/PriceFormater";

type Props = {
  //   children: React.ReactNode;
  title: string;
  subtitle: number;
  diff?: number;
  type?: "pricing";
};

const DashboardCard = (props: Props) => {
  return (
    <div
      className="flex flex-col rounded-lg bg-white dark:bg-slate-800 p-4
    shadow-lg"
    >
      <span className="text-slate-500">{props.title}</span>
      <div className="flex items-end">
        {props.type === "pricing" ? (
          <PriceFormater
            price={props.subtitle}
            className="text-xl font-bold"
            data-testid="pricing"
          />
        ) : (
          <span className="text-xl font-bold" data-testid="normal">
            {props.subtitle}
          </span>
        )}
        <div className="text-sm flex items-end ml-auto">
          {props.diff && (
            <>
              <span
                data-testid="diff"
                className={`${
                  props.diff > 0 ? "text-green-500" : "text-red-500"
                }`}
              >
                {Math.round((props.diff + Number.EPSILON) * 100) / 100}%
              </span>
              <span
                className={`${
                  props.diff > 0 ? "text-green-500" : "text-red-500"
                } animate-bounce`}
              >
                {props.diff > 0 ? (
                  <i
                    className="flex items-start fi-rr-angle-small-up"
                    data-testid="up"
                  ></i>
                ) : (
                  <i
                    className="flex items-start fi-rr-angle-small-down"
                    data-testid="down"
                  ></i>
                )}
              </span>
            </>
          )}
        </div>
      </div>
    </div>
  );
};

export default DashboardCard;
