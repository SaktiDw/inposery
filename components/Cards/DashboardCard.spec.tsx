import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import DashboardCard from "./DashboardCard";

describe("DashboardCard", () => {
  const tree = (
    title: string,
    subtitle: number,
    type?: "pricing" | undefined,
    diff?: number
  ) =>
    render(
      <DashboardCard
        title={title}
        subtitle={subtitle}
        type={type || undefined}
        diff={diff}
      />
    );

  it("should be rendered normal ", () => {
    const { container, queryByTestId } = tree("Today Profit", 100000);
    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
    expect(queryByTestId("normal")).toBeDefined();
    expect(queryByTestId("pricing")).toBeNull();
  });

  it("should be rendered pricing ", () => {
    const { container, queryByTestId } = tree(
      "Today Profit",
      100000,
      "pricing"
    );
    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
    expect(queryByTestId("normal")).toBeNull();
    expect(queryByTestId("pricing")).toBeDefined();
  });

  it("should be rendered the different from sales and modal", () => {
    const { container, queryByTestId } = tree(
      "Today Profit",
      100000,
      "pricing",
      9
    );
    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
    expect(queryByTestId("diff")).toBeDefined();
  });
  it("should be rendered the different with icon up", () => {
    const { container, queryByTestId } = tree(
      "Today Sales",
      100000,
      "pricing",
      9
    );
    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
    expect(queryByTestId("diff")).toBeDefined();
    expect(queryByTestId("up")).toBeDefined();
  });
  it("should be rendered the different with icon up", () => {
    const { container, queryByTestId } = tree(
      "Today Sales",
      100000,
      "pricing",
      -9
    );
    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
    expect(queryByTestId("diff")).toBeDefined();
    expect(queryByTestId("down")).toBeDefined();
  });
});
