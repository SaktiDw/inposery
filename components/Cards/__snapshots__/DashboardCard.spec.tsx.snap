// Vitest Snapshot v1

exports[`DashboardCard > should be rendered normal  1`] = `
<div>
  <div
    class="flex flex-col rounded-lg bg-white dark:bg-slate-800 p-4
 shadow-lg"
  >
    <span
      class="text-slate-500"
    >
      Today Profit
    </span>
    <div
      class="flex items-end"
    >
      <span
        class="text-xl font-bold"
        data-testid="normal"
      >
        100000
      </span>
      <div
        class="text-sm flex items-end ml-auto"
      />
    </div>
  </div>
</div>
`;

exports[`DashboardCard > should be rendered pricing  1`] = `
<div>
  <div
    class="flex flex-col rounded-lg bg-white dark:bg-slate-800 p-4
 shadow-lg"
  >
    <span
      class="text-slate-500"
    >
      Today Profit
    </span>
    <div
      class="flex items-end"
    >
      <span
        class="text-xl font-bold"
      >
        Rp 100.000
      </span>
      <div
        class="text-sm flex items-end ml-auto"
      />
    </div>
  </div>
</div>
`;

exports[`DashboardCard > should be rendered the different from sales and modal 1`] = `
<div>
  <div
    class="flex flex-col rounded-lg bg-white dark:bg-slate-800 p-4
 shadow-lg"
  >
    <span
      class="text-slate-500"
    >
      Today Profit
    </span>
    <div
      class="flex items-end"
    >
      <span
        class="text-xl font-bold"
      >
        Rp 100.000
      </span>
      <div
        class="text-sm flex items-end ml-auto"
      >
        <span
          class="text-green-500"
          data-testid="diff"
        >
          9
          %
        </span>
        <span
          class="text-green-500 animate-bounce"
        >
          <i
            class="flex items-start fi-rr-angle-small-up"
            data-testid="up"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`DashboardCard > should be rendered the different with icon up 1`] = `
<div>
  <div
    class="flex flex-col rounded-lg bg-white dark:bg-slate-800 p-4
 shadow-lg"
  >
    <span
      class="text-slate-500"
    >
      Today Sales
    </span>
    <div
      class="flex items-end"
    >
      <span
        class="text-xl font-bold"
      >
        Rp 100.000
      </span>
      <div
        class="text-sm flex items-end ml-auto"
      >
        <span
          class="text-green-500"
          data-testid="diff"
        >
          9
          %
        </span>
        <span
          class="text-green-500 animate-bounce"
        >
          <i
            class="flex items-start fi-rr-angle-small-up"
            data-testid="up"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`DashboardCard > should be rendered the different with icon up 2`] = `
<div>
  <div
    class="flex flex-col rounded-lg bg-white dark:bg-slate-800 p-4
 shadow-lg"
  >
    <span
      class="text-slate-500"
    >
      Today Sales
    </span>
    <div
      class="flex items-end"
    >
      <span
        class="text-xl font-bold"
      >
        Rp 100.000
      </span>
      <div
        class="text-sm flex items-end ml-auto"
      >
        <span
          class="text-red-500"
          data-testid="diff"
        >
          -9
          %
        </span>
        <span
          class="text-red-500 animate-bounce"
        >
          <i
            class="flex items-start fi-rr-angle-small-down"
            data-testid="down"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;
